File: scripts/export-collections.js
require('dotenv').config();
const { MongoClient } = require('mongodb');
const fs = require('fs');
const path = require('path');

async function exportCollections() {
  const uri = process.env.MONGODB_URI;
  const dbName = process.env.DB_NAME || 'lesson_booking';

  if (!uri) {
    throw new Error('MONGODB_URI is not set');
  }

  const client = new MongoClient(uri);

  try {
    await client.connect();
    console.log('Connected to MongoDB');

    const db = client.db(dbName);

    // Export lessons collection
    console.log('\nExporting lessons collection...');
    const lessons = await db.collection('lessons').find({}).toArray();
    const lessonsPath = path.join(__dirname, '..', 'exports', 'lessons.json');
    fs.writeFileSync(lessonsPath, JSON.stringify(lessons, null, 2));
    console.log(`✓ Exported ${lessons.length} lessons to ${lessonsPath}`);

    // Export orders collection
    console.log('\nExporting orders collection...');
    const orders = await db.collection('orders').find({}).toArray();
    const ordersPath = path.join(__dirname, '..', 'exports', 'orders.json');
    fs.writeFileSync(ordersPath, JSON.stringify(orders, null, 2));
    console.log(`✓ Exported ${orders.length} orders to ${ordersPath}`);

    console.log('\n✅ Export complete!');
    console.log('\nVerification:');
    console.log(`- Lessons: ${lessons.length} documents`);
    console.log(`- Orders: ${orders.length} documents`);

    if (lessons.length > 0) {
      console.log('\nSample lesson fields:', Object.keys(lessons[0]).join(', '));
    }
    if (orders.length > 0) {
      console.log('Sample order fields:', Object.keys(orders[0]).join(', '));
    }

  } catch (error) {
    console.error('Export failed:', error);
    process.exit(1);
  } finally {
    await client.close();
    console.log('\nMongoDB connection closed');
  }
}

exportCollections();

File: scripts/export-collections.py
#!/usr/bin/env python3
import os
import json
from pymongo import MongoClient
from urllib.parse import quote_plus

# MongoDB connection details
MONGODB_URI = "mongodb+srv://BAMHITS22:22310314@cst3144cw1.u3co8ym.mongodb.net/?appName=CST3144CW1"
DB_NAME = "lesson_booking"

def export_collections():
    try:
        # Connect to MongoDB
        print("Connecting to MongoDB...")
        client = MongoClient(MONGODB_URI)
        db = client[DB_NAME]

        # Export lessons collection
        print("\nExporting lessons collection...")
        lessons_cursor = db['lessons'].find({})
        lessons = list(lessons_cursor)

        # Convert ObjectId to string for JSON serialization
        for lesson in lessons:
            if '_id' in lesson:
                lesson['_id'] = str(lesson['_id'])

        lessons_path = os.path.join(os.path.dirname(__file__), '..', 'exports', 'lessons.json')
        with open(lessons_path, 'w') as f:
            json.dump(lessons, f, indent=2, default=str)
        print(f"✓ Exported {len(lessons)} lessons to {lessons_path}")

        # Export orders collection
        print("\nExporting orders collection...")
        orders_cursor = db['orders'].find({})
        orders = list(orders_cursor)

        # Convert ObjectId to string for JSON serialization
        for order in orders:
            if '_id' in order:
                order['_id'] = str(order['_id'])
            if 'lessonIDs' in order:
                order['lessonIDs'] = [str(lid) for lid in order['lessonIDs']]
            if 'createdAt' in order:
                order['createdAt'] = str(order['createdAt'])

        orders_path = os.path.join(os.path.dirname(__file__), '..', 'exports', 'orders.json')
        with open(orders_path, 'w') as f:
            json.dump(orders, f, indent=2, default=str)
        print(f"✓ Exported {len(orders)} orders to {orders_path}")

        print("\n✅ Export complete!")
        print(f"\nVerification:")
        print(f"- Lessons: {len(lessons)} documents")
        print(f"- Orders: {len(orders)} documents")

        if lessons:
            print(f"\nSample lesson fields: {', '.join(lessons[0].keys())}")
        if orders:
            print(f"Sample order fields: {', '.join(orders[0].keys())}")

        client.close()
        print("\nMongoDB connection closed")

    except Exception as e:
        print(f"Export failed: {e}")
        exit(1)

if __name__ == "__main__":
    export_collections()
